garak/generators/huggingface.py

```diff
@@ -493,15 +493,11 @@ def _call_model(self, prompt: str, generations_this_call: int = 1):
                     prompt, truncation=True, return_tensors="pt"
                 ).to(self.init_device)
 
-                try:
-                    outputs = self.model.generate(
-                        **inputs, generation_config=self.generation_config
-                    )
-                except IndexError as e:
-                    if len(prompt) == 0:
-                        return [""] * generations_this_call
-                    else:
-                        raise e
+                if len(prompt) == 0:
+                    return [""] * generations_this_call
+                outputs = self.model.generate(
+                    **inputs, generation_config=self.generation_config
+                )
```

Comment:
This return could also me moved up before the `self.tokenizer` call for `inputs`.

Reply:
Some models can provide interesting outputs when prompted with nothing; it's a shame that others lead to exceptions being raised under the same conditions. Prefer try-catch. I'm not sure the `transformers` lib gives us any guarantees about the kind of exception raised, which is unfortunate.

Reply:
Fair, if `transformers` does not offer a specific error type we can catch `Exception` and log any exception from `generate` for an empty prompt and return a set of `None`. As noted in that comments on that review these responses would currently be filtered for detector check, however that seems like a reasonable compromise if we log the error.

Maybe this merits a separate PR to help follow the breadcrumbs in the future.

Reply:
Addressed as #690